koku/api/report/ocp/query_handler.py

```diff
@@ -326,80 +263,27 @@ def execute_query(self):  # noqa: C901
         self.query_data = data
         return self._format_query_response()
 
+    # Capacity Calculations
+
     def get_capacity(self, query_data):  # noqa: C901
-        """Calculate cluster capacity for all nodes over the date range."""
-        if is_grouped_by_node(self.parameters):
-            _capacity = self._generate_capacity_subsets("node", "node")
-        else:
-            _capacity = self._generate_capacity_subsets("cluster", "cluster_id")
-        if not _capacity:
-            return query_data, {}
-
-        if _capacity.level_key == "node":
-            for row in query_data:
-                row[_capacity.key] = _capacity.resolution_level_total.get(row.get("date"), {}).get(
-                    row.get("node"), Decimal(0)
-                )
-                _calculate_unused(row)
-            return query_data, _capacity.generate_query_sum()
-        else:
-            return self._get_cluster_capacity(query_data, _capacity)
-
-    def _generate_capacity_subsets(self, capacity_level, group_by):
-        """Calculate capacity over the timescope and resolution"""
-        _capacity = None
-        annotations = self._mapper.report_type_map.get("capacity_aggregate", {}).get(capacity_level)
-        if not annotations:
-            return None
-        cap_key = list(annotations.keys())[0]
+        """Calculate capacity & instance count for all nodes over the date range."""
         q_table = self._mapper.query_table
         LOG.debug(f"Using query table: {q_table}")
         query = q_table.objects.filter(self.query_filter)
         if self.query_exclusions:
             query = query.exclude(self.query_exclusions)
         with tenant_context(self.tenant):
-            cap_data = query.values(*["usage_start", group_by]).annotate(**annotations)
-            for entry in cap_data:
-                if not _capacity:
-                    _capacity = CapacitySubsets(
-                        key=cap_key, level_key=capacity_level, count_units=entry.get("capacity_count_units")
-                    )
-                level_value = entry.get(capacity_level, "")
-                usage_start = entry.get("usage_start", "")
-                if self.resolution == "daily" and isinstance(usage_start, datetime.date):
-                    usage_start = usage_start.isoformat()
-                if self.resolution == "monthly":
-                    usage_start = usage_start.strftime("%Y-%m")
-                cap_value = entry.get(cap_key, 0)
-                _capacity.add(cap_value, usage_start, level_value)
-                _capacity.add_instance_count(entry.get("capacity_count"), usage_start, level_value)
-        return _capacity
-
-    def _get_cluster_capacity(self, query_data, _capacity):
-        """Calculate the cluster capacity."""
+            _class = NodeCapacity if is_grouped_by_node(self.parameters) else ClusterCapacity
```

Comment:
What if we are `filter`ing by Node?

Reply:
We have only ever shown node capacity when grouping by node, it was the requirement for the original epic.